amundsen_application/static/js/components/TableDetail/RequestMetadataForm/tests/index.spec.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+
+import { shallow } from 'enzyme';
+
+import { RequestMetadataForm, mapDispatchToProps, mapStateToProps, RequestMetadataProps } from '../';
+import globalState from 'fixtures/globalState';
+import { NotificationType } from 'interfaces/';
+
+const mockFormData = { 
+  'recipients': 'test1@test.com,test2@test.com',
+  'sender': 'test@test.com',
+  'table-description': 'on',
+  'fields-requested': 'off',
+  'comment': 'test',
+  get: jest.fn(),
+}
+mockFormData.get.mockImplementation((val) => {
+  return mockFormData[val];
+})
+// @ts-ignore: How to mock FormData without TypeScript error?
+global.FormData = () => (mockFormData);
+
+describe('RequestMetadataForm', () => {
+  const setup = () => {
+    const props: RequestMetadataProps = {
+      userEmail: '',
+      tableName: '',
+      tableOwners: [''],
+      submitNotification: jest.fn(),
+      requestIsOpen: false,
+      toggleRequest: jest.fn(),
+    };
+    const wrapper = shallow<RequestMetadataForm>(<RequestMetadataForm {...props} />)
+    return {props, wrapper}
+  };
+
+  describe('toggle', () => {
+    it('calls toggleRequest', () => {
+      const { props, wrapper } = setup();
+      const toggleRequestSpy = jest.spyOn(props, 'toggleRequest');
+      wrapper.instance().toggle();
+      expect(toggleRequestSpy).toHaveBeenCalled();
+    });
+  });
+
+  describe('submitNotification', () => {
+    it('calls submitNotification', () => {
+      const { props, wrapper } = setup();
+      const submitNotificationSpy = jest.spyOn(props, 'submitNotification');
+      wrapper.instance().submitNotification({ preventDefault: jest.fn() });
+      expect(submitNotificationSpy).toHaveBeenCalledWith(
+        mockFormData['recipients'].split(','),
+        mockFormData['sender'],
+        NotificationType.REQUESTED,
+        {
+          comment: mockFormData['comment'],
+          resource_name: props.tableName,
+          resource_url: window.location.href,
+          description_requested: mockFormData['table-description'] === "on" ? true : false,
+          fields_requested: mockFormData['fields-requested'] === "on" ? true : false,
+        }
+      );
+    });
+  });
+  
+  // TODO
+  describe('render', () => {
+  });
+
+  describe('mapStateToProps', () => {
+    let result;
+    beforeAll(() => {
+      result = mapStateToProps(globalState);
+    });
+
+    it('sets userEmail on the props', () => {
+      expect(result.userEmail).toEqual(globalState.user.loggedInUser.email);
+    });
+    it('sets tableName on the props', () => {
+      expect(result.tableName).toEqual(globalState.tableMetadata.tableData.schema + '.' + globalState.tableMetadata.tableData.table_name);
```

Comment:
nit: ideally we'd have a method that generates this value so that the test isn't dependent on executing extra logic. Not need to address this now but in the future it will be best to have a test like:
```
expect(result.tableName).toEqual(methodThatOwnsThisLogic(...));
```

That way, if we ever change how the tableName is generated, we won't have to change the test itself.